data_structures/hash_table/hash_table.py

```diff
@@ -11,6 +11,17 @@ def __init__(self, max_size=1024):
         self.max_size = max_size
         self.buckets = [LinkedList() for _ in range(max_size)]
 
+    def __iter__(self):
+        """Allow the hash table to be iterable."""
+        iterable = []
+        for bucket in self.buckets:
+            node = bucket.head
+            while node:
+                for key in node.val.keys():
+                    iterable.append(key)
+                node = node._next
+        return iter(iterable)
```

Comment:
Just so you know, you can implement this as a generator instead so that you do not create a list of all the keys in the `HashTable`.